x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/operator/OperatorPrivilegesSingleNodeTests.java

```diff
@@ -0,0 +1,286 @@
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.util.set.Sets;
+import org.elasticsearch.test.SecuritySingleNodeTestCase;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.elasticsearch.test.SecuritySettingsSource.TEST_PASSWORD_HASHED;
+import static org.elasticsearch.test.SecuritySettingsSourceField.TEST_PASSWORD;
+import static org.elasticsearch.xpack.core.security.authc.support.UsernamePasswordToken.basicAuthHeaderValue;
+import static org.hamcrest.Matchers.containsString;
+
+public class OperatorPrivilegesSingleNodeTests extends SecuritySingleNodeTestCase {
+
+    private static final String OPERATOR_USER_NAME = "test_operator";
+
+    @Override
+    protected String configUsers() {
+        return super.configUsers()
+            + OPERATOR_USER_NAME + ":" + TEST_PASSWORD_HASHED + "\n";
+    }
+
+    @Override
+    protected String configRoles() {
+        return super.configRoles()
+            + "limited_operator:\n"
+            + "  cluster:\n"
+            + "    - 'cluster:admin/voting_config/clear_exclusions'\n"
+            + "    - 'monitor'\n";
+    }
+
+    @Override
+    protected String configUsersRoles() {
+        return super.configUsersRoles()
+            + "limited_operator:" + OPERATOR_USER_NAME + "\n";
+    }
+
+    @Override
+    protected String configOperatorUsers() {
+        return super.configOperatorUsers()
+            + "operator:\n"
+            + "  - usernames: ['" + OPERATOR_USER_NAME + "']\n";
+    }
+
+    @Override
+    protected Settings nodeSettings() {
+        Settings.Builder builder = Settings.builder().put(super.nodeSettings());
+        // Ensure the new settings can be configured
+        builder.put("xpack.security.operator_privileges.enabled", "true");
+        return builder.build();
+    }
+
+    // TODO: Not all plugins are available in internal cluster tests. Hence not all action names can be checked.
+    public void testActionsAreEitherOperatorOnlyOrNot() {
+        final Injector injector = node().injector();
+        final List<Binding<TransportAction>> bindings = injector.findBindingsByType(TypeLiteral.get(TransportAction.class));
+
+        final List<String> allActionNames = new ArrayList<>(bindings.size());
+        for (final Binding<TransportAction> binding : bindings) {
+            allActionNames.add(binding.getProvider().get().actionName);
+        }
+
+        final Set<String> nonOperatorActions = Set.of(NON_OPERATOR_ACTIONS);
+        final Set<String> expectedOperatorOnlyActions = Sets.difference(Set.copyOf(allActionNames), nonOperatorActions);
+        final Set<String> actualOperatorOnlyActions = new HashSet<>(CompositeOperatorOnly.ActionOperatorOnly.SIMPLE_ACTIONS);
+        assertTrue(actualOperatorOnlyActions.containsAll(expectedOperatorOnlyActions));
+        assertFalse(actualOperatorOnlyActions.removeAll(nonOperatorActions));
```

Comment:
Sorry there are some messiness in the code because I tried different ideas and didn't clean it up after settled. This is the earlier attempt for ensuring every action is marked as either operator or non-operator. However, as discussed, SingleNodeTest is not the right approach because it does not cover all xpack plugins. I have since added the test with an external cluster + plugin test (OpTestPlugin, it's a bad name and I'll work on it). So I will remove all these code in SingleNodeTest and turn it into just a feature tests, i.e. whether the feature works or not. So in summary, we will have:
* SingleNodeTest for features
* External cluster + plugin for testing "action completeness" and also features. There will be some overlap in terms of feature testing. But I think it also provides addtional value cocmpared to a SingleNodeTest, e.g. cross-nodes, actual file reading, and settings via yml config.
* Other unit tests.